jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/DMatrix.java

```diff
@@ -388,13 +385,68 @@ public float[] getWeight() throws XGBoostError {
     return getFloatInfo("weight");
   }
 
+  /**
+   * Set weight of DMatrix from array interface
+   *
+   * @param column the XGBoost Column to provide the array interface
+   *               of weight column
+   * @throws XGBoostError native error
+   */
+  public void setWeight(Column column) throws XGBoostError {
+    setXGBDMatrixInfo("weight", column.getArrayInterfaceJson());
+  }
+
+  /**
+   * set weight of each instance
+   *
+   * @param weights weights
+   * @throws XGBoostError native error
+   */
+  public void setWeight(float[] weights) throws XGBoostError {
+    XGBoostJNI.checkCall(XGBoostJNI.XGDMatrixSetFloatInfo(handle, "weight", weights));
+  }
+
   /**
    * Get base margin of the DMatrix.
    */
   public float[] getBaseMargin() throws XGBoostError {
     return getFloatInfo("base_margin");
   }
 
+  /**
+   * Set base margin of DMatrix from array interface
+   *
+   * @param column the XGBoost Column to provide the array interface
+   *               of base margin column
+   * @throws XGBoostError native error
+   */
+  public void setBaseMargin(Column column) throws XGBoostError {
+    setXGBDMatrixInfo("base_margin", column.getArrayInterfaceJson());
+  }
+
+  /**
+   * Set base margin (initial prediction).
+   * <p>
+   * The margin must have the same number of elements as the number of
+   * rows in this matrix.
+   */
+  public void setBaseMargin(float[] baseMargin) throws XGBoostError {
```

Comment:
Base margin can be a matrix for multi-class classification.

Reply:
Yeah, That's true. the issue should have been there for days. This PR didn't mean to try to fix it. so let's just keep it here